File: 프로그래머스/lv2/42888. 오픈채팅방/오픈채팅방.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.HashMap;
import java.util.List;
class Solution {
    public String[] solution(String[] record) {
       HashMap<String, String> id = new HashMap<>();
        for(String s : record){
            if(!s.split(" ")[0].equals("Leave")){
                id.put(s.split(" ")[1], s.split(" ")[2]);
            }
        }

        List<String> arr = new ArrayList<String>();
        for(String s : record){
            if(s.split(" ")[0].equals("Enter")){
                String target = id.get(s.split(" ")[1]) + "님이 들어왔습니다.";
                arr.add(target);
            }   
            else if(s.split(" ")[0].equals("Leave")){
                String target = id.get(s.split(" ")[1]) + "님이 나갔습니다.";
                arr.add(target);
            }
        }
        String[] answer;
        answer = new String[arr.size()];
        arr.toArray(answer);
        
        return answer;
    }
}

File: 프로그래머스/2/42888. 오픈채팅방/오픈채팅방.js
function solution(record) {
    const answer = [];
    
    const map = new Map(); // 최종 닉네임
    for (const item of record) {
        const [type, id, nickname] = item.split(' ');
        if (type === 'Leave') continue;
        map.set(id, nickname);
    }
    
    for (const item of record) {
        const [type, id, nickname] = item.split(' ');
        
        if (type === 'Enter') {
            answer.push(`${map.get(id)}님이 들어왔습니다.`);
        } else if (type === 'Leave'){
            answer.push(`${map.get(id)}님이 나갔습니다.`)
        }
    }
    
    return answer;
}